pkg/bigquery/mapping_table.go

```diff
@@ -18,48 +18,111 @@ type MappingTableManager struct {
 	ctx              context.Context
 	mappingTableName string
 	client           *Client
+	schema           bigquery.Schema
 }
 
-func NewMappingTableManager(ctx context.Context, client *Client, mappingTable string) *MappingTableManager {
-	return &MappingTableManager{
-		ctx:              ctx,
-		mappingTableName: mappingTable,
-		client:           client,
+type TestMappingTableManager struct {
+	MappingTableManager
+}
+
+func NewTestMappingTableManager(ctx context.Context, client *Client, mappingTable string, schema bigquery.Schema) *TestMappingTableManager {
+	return &TestMappingTableManager{
+		MappingTableManager{
+			ctx:              ctx,
+			mappingTableName: mappingTable,
+			client:           client,
+			schema:           schema,
+		},
 	}
 }
 
-func (tm *MappingTableManager) Migrate() error {
-	dataset := tm.client.bigquery.Dataset(tm.client.datasetName)
-	table := dataset.Table(tm.mappingTableName)
+type VariantMappingTableManager struct {
+	MappingTableManager
+}
+
+func NewVariantMappingTableManager(ctx context.Context, client *Client, mappingTable string, schema bigquery.Schema) *VariantMappingTableManager {
```

Comment:
Do you have a prune implementation? I would work the same way as the other mappings table.  

You can't delete or update rows in BigQuery immediately after streaming inserts (due to the streaming buffer), using a "prune and append" approach helps maintain data consistency without waiting for the buffer to flush.

Essentially:
1. Remove anything but the latest set of inserts (prune)
2. Insert new records

The table is never empty, there's always something to query. You will end up having 2 copies of the data in the table, but you just query on max(created_at).

Reply:
I see how the current implementation is now. This is a little different than what I had in mind. I thought variant mappings should be relatively static and therefore I tried to avoid updating all entries whenever the command is run. Instead, I pull the existing entries and compare to see if there is new ones. If there is, I only insert the new entries with the latest created_at. That means my entries are not all created at the same time and it will be not pruned well with the current mechanism. Now I wonder if my way is even worth it.